system/lib/compiler-rt/lib/asan/asan_poisoning.h

```diff
@@ -54,6 +54,8 @@ ALWAYS_INLINE void FastPoisonShadow(uptr aligned_beg, uptr aligned_size,
       // RTEMS doesn't have have pages, let alone a fast way to zero
       // them, so default to memset.
       SANITIZER_RTEMS == 1 ||
+      // Emscripten doesn't have a nice way to zero ranges
+      SANITIZER_EMSCRIPTEN == 1 ||
       shadow_end - shadow_beg < common_flags()->clear_shadow_mmap_threshold) {
     REAL(memset)((void*)shadow_beg, value, shadow_end - shadow_beg);
```

Comment:
I wonder if it's worth doing a bulk-memory enabled build to get `memory.fill` instructions here.

Reply:
The builtin memset should be made to use `memory.fill` instructions, if it doesn't do so already. The "fast" way here refers to using page table magic to zero pages, which we don't have. We just need a fast memset.